src/subdomains/supporting/payout/strategies/payout/impl/lightning.strategy.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable } from '@nestjs/common';
+import { Blockchain } from 'src/integration/blockchain/shared/enums/blockchain.enum';
+import { LightningAddressType, LightningHelper } from 'src/integration/lightning/lightning-helper';
+import { Asset, AssetType } from 'src/shared/models/asset/asset.entity';
+import { AssetService } from 'src/shared/models/asset/asset.service';
+import { DfxLogger } from 'src/shared/services/dfx-logger';
+import { PayoutOrder } from '../../../entities/payout-order.entity';
+import { FeeResult } from '../../../interfaces';
+import { PayoutOrderRepository } from '../../../repositories/payout-order.repository';
+import { PayoutLightningService } from '../../../services/payout-lightning.service';
+import { PayoutStrategy } from './base/payout.strategy';
+
+@Injectable()
+export class LightningStrategy extends PayoutStrategy {
+  private readonly logger = new DfxLogger(LightningStrategy);
+
+  constructor(
+    private readonly assetService: AssetService,
+    private readonly payoutLightningService: PayoutLightningService,
+    private readonly payoutOrderRepo: PayoutOrderRepository,
+  ) {
+    super();
+  }
+
+  get blockchain(): Blockchain {
+    return Blockchain.LIGHTNING;
+  }
+
+  get assetType(): AssetType {
+    return undefined;
+  }
+
+  async doPayout(orders: PayoutOrder[]): Promise<void> {
+    if (await this.isHealthy()) {
+      for (const order of orders) {
+        try {
+          const address = order.destinationAddress;
+          const amount = order.amount;
+
+          const addressType = LightningHelper.getAddressType(address);
+
+          switch (addressType) {
+            case LightningAddressType.LN_URL: {
+              const txId = await this.payoutLightningService.sendPaymentByLnurl(address, amount);
+              await this.finishDoPayout(order, txId);
+              break;
+            }
+
+            case LightningAddressType.LN_NID: {
+              const txId = await this.payoutLightningService.sendPaymentByLnnid(address, amount);
+              await this.finishDoPayout(order, txId);
+              break;
+            }
+
+            case LightningAddressType.LND_HUB: {
+              const txId = await this.payoutLightningService.sendPaymentByLndhub(address, amount);
+              await this.finishDoPayout(order, txId);
+              break;
+            }
+
+            default:
+              this.logger.error(`Unknown address type ${addressType} in Lightning payout order ${order.id}`);
+          }

```

Comment:
I think it could make sense to move this to the PayoutLightningService, it would reduce the code.